SEDaily-IOS/GeneralCollectionViewController.swift

```diff
@@ -18,281 +16,144 @@ class GeneralCollectionViewController: UICollectionViewController {
-                if hasChanges {
-                    self.loading = false
-                }
-                
-                self.data = PodcastModel.getTop()
-                self.registerNotifications()
-            })
-            break
-        default:
-            break
-        }
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
     }
-}
 
-extension GeneralCollectionViewController {
     // MARK: UICollectionViewDataSource
-    
+
     override func numberOfSections(in collectionView: UICollectionView) -> Int {
         // #warning Incomplete implementation, return the number of sections
         return 1
     }
-    
+
+
     override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        if itemCount > 0 {
+        if podcastViewModelController.viewModelsCount > 0 {
             self.skeletonCollectionView.fadeOut(duration: 0.5, completion: nil)
         }
-        if itemCount < 10 {
-            self.getData()
-        }
-        return itemCount
+        return podcastViewModelController.viewModelsCount
     }
-    
+
     override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! PodcastCell
-        
-        let item = data[indexPath.row]
-        
-        checkPage(indexPath: indexPath, item: item)
-        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! PodcastCell
+    
         // Configure the cell
-        let uploadDate = Date(iso8601String: (item.uploadDate ?? ""))
-        cell.setupCell(imageURLString: item.imageURLString, title: item.podcastName!, timeLength: nil, date: uploadDate)
-
+        if let viewModel = podcastViewModelController.viewModel(at: indexPath.row) {
+            cell.viewModel = viewModel
+            if let lastIndexPath = self.collectionView?.indexPathForLastItem {
+                if let lastItem = podcastViewModelController.viewModel(at: lastIndexPath.row) {
+                    self.checkPage(currentIndexPath: indexPath,
+                                   lastIndexPath: lastIndexPath,
+                                   lastIdentifier: lastItem.uploadDateiso8601)
+                }
+            }
+        }
+    
         return cell
     }
     
-    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        let item = data[indexPath.row]
-        let vc = PostDetailTableViewController()
-        vc.model = item
-        self.navigationController?.pushViewController(vc, animated: true)
-    }
-}
+    var loading = false
+    let pageSize = 10
```

Comment:
Prefer to gather all the member variables together at the top. If they are local to the function, then put them inside the function.